src/hotspot/os/windows/os_windows.cpp

```diff
@@ -5249,7 +5249,18 @@ class HighResolutionInterval : public CHeapObj<mtThread> {
 // explicit "PARKED" == 01b and "SIGNALED" == 10b bits.
 //
 
-int PlatformEvent::park(jlong Millis) {
+int PlatformEvent::park_nanos(jlong nanos) {
+  assert(0 <= nanos, "nanos are in range");
+
+  // Windows timers are still quite unpredictable to handle sub-millisecond granularity.
+  // Instead of implementing this method, fall back to the millisecond sleep, treating
+  // any positive requested nanos as a full millisecond.
+  jlong millis = align_up(nanos, NANOSECS_PER_MILLISEC) / NANOSECS_PER_MILLISEC;
+  assert(nanos == 0 || millis != 0, "Only pass zero millis on zero nanos");
```

Comment:
Right, I thought we are allowed to pass zero downstream, but there is a `guarantee(Millis > 0)`, which would fail. So there is no reason to do this. Fixed.